.github/workflows/build-in-docker.yml

```diff
@@ -7,32 +7,50 @@ on:
     branches: [ "dev", "candidate", "release", "jshooks" ]
 
 concurrency:
-  group: ${{ github.workflow }}
-  cancel-in-progress: false
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
   checkout:
     runs-on: [self-hosted, vanity]
     steps:
-    - uses: actions/checkout@v4
-      with:
-        clean: false
+      - uses: actions/checkout@v4
+        with:
+          path: ${{ github.sha }}
+          clean: true  # ensure clean checkout
```

Comment:
AI drivel comment 

It's going to be a per commit unique path anyway ...

Reply:
Although, I do wonder about what conditions could cause a clobbering here. I'm not familiar enough with self-hosted runners and how concurrency works. I guess you could push the same commit at the top of different branches.